usecase/parser/msg.go

```diff
@@ -136,15 +136,27 @@ func ParseBlockResultsTxsMsgToCommands(
 			case "/ibc.applications.transfer.v1.MsgTransfer":
 				msgCommands = ibcmsg.ParseMsgTransfer(msgCommonParams, txsResult, msgIndex, msg)
 			case "/ibc.core.channel.v1.MsgRecvPacket":
-				msgCommands = ibcmsg.ParseMsgRecvPacket(msgCommonParams, txsResult, msgIndex, msg)
+				parser := parserManager.GetParser(event.MSG_IBC_RECV_PACKET, utils.ParserBlockHeight(blockHeight))
+				msgCommands = parser(utils.CosmosParserParams{
+					MsgCommonParams: msgCommonParams,
+					Msg: msg,
+					MsgIndex: msgIndex,
+					TxsResult: txsResult,
+				})
 			case "/ibc.core.channel.v1.MsgAcknowledgement":
 				msgCommands = ibcmsg.ParseMsgAcknowledgement(msgCommonParams, txsResult, msgIndex, msg)
 			case "/ibc.core.channel.v1.MsgTimeout":
 				msgCommands = ibcmsg.ParseMsgTimeout(msgCommonParams, txsResult, msgIndex, msg)
 			case "/ibc.core.channel.v1.MsgTimeoutOnClose":
 				msgCommands = ibcmsg.ParseMsgTimeoutOnClose(msgCommonParams, txsResult, msgIndex, msg)
 			case "/cosmos.authz.v1beta1.MsgGrant":
-				msgCommands = parseMsgGrant(msgCommonParams, msg)
+				parser := parserManager.GetParser(event.MSG_GRANT, utils.ParserBlockHeight(blockHeight))
```

Comment:
A suggestion after more message parsers are updated:

- Keep a mapping of `Cosmos type string -> event type` i.e. ["/cosmos.authz.v1beta1.MsgGrant"] -> event.MSG_GRANT
```
eventType, _ = CosmosMessageTypeEventMap[msg["@type"]
parser := parserManager.GetParser(eventType utils.ParserBlockHeight(blockHeight))
msgCommands = parser(utils.CosmosParserParams{
	MsgCommonParams: msgCommonParams,
	Msg: msg,
	MsgIndex: msgIndex,
	TxsResult: txsResult,
})
```

Reply:
How about this?

```
func GetParserKeyFromMsgType(msgType string) CosmosParserKey {
	elements := strings.Split(msgType, ".")

	return CosmosParserKey(elements[len(elements)-1])
}
```

Reply:
okay, could do in this way first